subsys/bluetooth/mesh/transport.c

```diff
@@ -1001,20 +1016,57 @@ static int ctl_recv(struct bt_mesh_net_rx *rx, uint8_t hdr,
 			return bt_mesh_lpn_friend_clear_cfm(rx, buf);
 		}
 
-		if (!rx->friend_cred) {
-			LOG_WRN("Message from friend with wrong credentials");
-			return -EINVAL;
-		}
-
 		switch (ctl_op) {
 		case TRANS_CTL_OP_FRIEND_UPDATE:
+			if (rx->ctx.cred != BT_MESH_CRED_FRIEND) {
```

Comment:
I guess because the rx address can be the lpn's address, but the opcode not necessarily Friend Update or Subscription Confirmation. But what about this:
```
if ((ctl_op == TRANS_CTL_OP_FRIEND_UPDATE ||
     ctl_op == TRANS_CTL_OP_FRIEND_SUB_CFM) &&
     rx->ctx.cred != BT_MESH_CRED_FRIEND) {
	LOG_WRN("Message from friend with wrong credentials");
	return -EINVAL;
}
```
this will preserve the error message. Otherwise it could be just:
```
if (rx->friend_cred) {
	switch (ctl_op) {
	// TRANS_CTL_OP_FRIEND_UPDATE 
	// TRANS_CTL_OP_FRIEND_SUB_CFM
...
```